File: hardware_broadcast/logger/logger.go
package logger

import (
	"os"
	"time"

	"github.com/rs/zerolog"
	"github.com/rs/zerolog/log"
)

func InitLogger() {
	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.Kitchen})
	log.Info().Msg("Logger inted")
}

func LogInformation(message string) {
	log.Info().Msg(message)
}
func LogError(err error) {
	log.Error().Msg(err.Error())
}
func LogFatal(err error) {
	log.Fatal().Msg(err.Error())
}

File: HardwareBroadcast/Logging/Logger.cs
namespace HardwareBroadcast.Logging;

public class Logger : ILogger
{
    private Serilog.Core.Logger _logger { get; }

    public Logger(Serilog.Core.Logger logger)
    {
        _logger = logger;
    }
    
    public void LogInformation(string message)
    {
        _logger.Information(message);
    }

    public void LogWarning(string message)
    {
        _logger.Warning(message);
    }

    public void LogError(string message)
    {
        _logger.Error(message);
    }

    public void LogError(Exception exception)
    {
        _logger.Error(exception, exception.Message);
    }

    public void LogFatal(string message)
    {
        _logger.Fatal(message);
    }

    public void LogFatal(Exception exception)
    {
        _logger.Fatal(exception, exception.Message);
    }
}